src/runtime/runtime.cs

```diff
@@ -148,7 +149,8 @@ internal static void Initialize(bool initSigs = false, ShutdownMode mode = Shutd
             }
             else
             {
-                PyGILState_Ensure();
+                PyEval_InitThreads();
```

Comment:
This needs some explanation. Also, according to [this SO](https://stackoverflow.com/questions/15470367/pyeval-initthreads-in-python-3-how-when-to-call-it-the-saga-continues-ad-naus), `PyEval_InitThreads` acquires GIL, so I am not sure what to think about `PyGILState_Ensure` in the next line.

Reply:
This change will occur a bug by removing the `PyGILState_Ensure`.
GIL released on the previous shutdown, it needs GIL here.
`PyEval_InitThreads` is not the necessary one, it initialized at previous startup.

Reply:
Looking at Python's code (ceval.c), `PyEval_InitThreads` only acquires the GIL if the GIL hasn't been created yet (also creates it). Calling it more than once is a no-op, and does not acquire it. `PyGILState_Ensure` is required so that the GIL is effectively acquired. @amos402 is right that the `PyEval_InitThreads` call is not required. I'll remove it and add a code comment.